packages/blockchain/src/index.ts

```diff
@@ -478,6 +478,9 @@ export default class Blockchain implements BlockchainInterface {
     // construct recent block signers list with this block
     let signers = this._cliqueLatestBlockSigners
     signers = signers.slice(signers.length < limit ? 0 : 1)
+    if (signers.length > 0 && !signers[signers.length - 1][0].eq(header.number.subn(1))) {
+      signers = []
```

Comment:
Should we just return false here? With an explanatory comment like `// if the last signed block is not one from the current head we do not have a complete picture of the state to verify if too recently signed`

Reply:
Yes, guess that's the same, right?

Reply:
(but definitely better)

Reply:
Can you reapprove here?